cafebabel/users/views.py

```diff
@@ -41,10 +42,20 @@ def edit(id):
         if request.form.get('delete'):
             user.profile.delete_image()
         if image:
-            if file_exceeds(image,
-                            current_app.config['USERS_IMAGE_MAX_LENGTH']):
+            maximum = current_app.config['USERS_IMAGE_MAX_CONTENT_LENGTH']
+            if file_exceeds(image, maximum):
                 raise exceptions.RequestEntityTooLarge()
-            user.profile.attach_image(image)
+            if image.filename == '':
+                flash('There was an error in your profile submission:')
+                flash('No selected file.')
+                return redirect(url_for('users.edit', id=user.id))
+            if allowed_file(image.filename):
```

Comment:
as before I guess it would look more straightforward to read
```python
if not allowed_file(image.filename):
    flash('There was an error in your profile submission:')
    flash('Unallowed extension.')
    return redirect(url_for('users.edit', id=user.id))
user.profile.attach_image(image)
```